# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxUserFontSet.h
# Commit: bd78ccc10d70
# Full Hash: bd78ccc10d704a29bccc83a3edd977cd99f28027
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-03-27 09:43:11
# Regressor Bug: 1694123
# File Overlap Count: 1
# Description:
#   Bug 1694123 - Ensure user fonts with src:local() are refreshed after a font-list update, even if the local font lookup previously failed. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106511
# ==============================================================================

diff -r af4bfd54ecd0 -r bd78ccc10d70 gfx/thebes/gfxUserFontSet.h
--- a/gfx/thebes/gfxUserFontSet.h	Fri Mar 26 16:03:33 2021 +0000
+++ b/gfx/thebes/gfxUserFontSet.h	Fri Mar 26 16:04:38 2021 +0000
@@ -614,6 +614,8 @@
     mUserFontLoadState = STATUS_NOT_LOADED;
     mFontDataLoadingState = NOT_LOADING;
     mLoader = nullptr;
+    mSrcIndex = 0;
+    mSeenLocalSource = false;
   }
 
   // whether to wait before using fallback font or not
@@ -764,6 +766,7 @@
   };
   FontDataLoadingState mFontDataLoadingState;
 
+  bool mSeenLocalSource;
   bool mUnsupportedFormat;
   mozilla::StyleFontDisplay mFontDisplay;  // timing of userfont fallback
 
